bolt/test/AArch64/lite-mode.s

```diff
@@ -0,0 +1,109 @@
+## Check that in lite mode llvm-bolt updates function references in
+## non-optimized code.
+
+# RUN: llvm-mc -filetype=obj -triple aarch64-unknown-unknown %s -o %t.o
+# RUN: link_fdata %s %t.o %t.fdata
+# RUN: llvm-strip --strip-unneeded %t.o
+# RUN: %clang %cflags %t.o -o %t.exe -Wl,-q -static
+# RUN: llvm-bolt %t.exe -o %t.bolt --data %t.fdata --lite \
+# RUN:   --compact-code-model
```

Comment:
I know we'll eventually move away from a relaxation method that isn't based on assumptions, but since LongJmp is currently the default path, it might be worth duplicating the checks for now?

eg, testing another bolted binary that doesn't rely on the compact-code model

Reply:
I'll remove that option since we don't really need it for the test case.